pkg/version/version.go

```diff
@@ -15,8 +15,10 @@
 // Package version contains the version information of the krew binary.
 package version
 
+const unknown = "unknown"
```

Comment:
I don't know why this is a linter error. It’s pretty common coding practice to have package level-`const`s that are used throughout the code rather than duplicating it.

Another example is below. We should probably turn off this linter, it’s too aggressive.

Reply:
I don't quite understand what you mean. Travis and my machine think that the linter passes.

Do you mean that the constant should _not_ be there and be inlined instead? This is what the linter complained about so I extracted the constants. If you think this is bad, I can see what linter needs to be disabled.

Reply:
ah, wait, I now see that the linter is actually trying to help us.

that said, it's erroneous. the "unknown" string on both gitCommit() and gitTag() methods mean different things. so they should not be shared by the same variable. we can do:


    const unknownGitCommit = "unknown"
    const unknownGitTag = "unknown"

similarly, the tests below actually test against `v != unknown` which isn't good.

When we update the value of the const, we might be breaking the API (in this case we aren't), but the if test continues to test `v != "unknown"` literal string comparison, it could catch that.

So I think we should still disable this particular linter.

Reply:
good catch. I disabled the `goconst` linter. I think it would have allowed two occurrences of "unknown" in tests, but `goconst` is not critical at all.

Reply:
@ahmetb can you take another look or close this PR?